core/Dispatch.cpp

```diff
@@ -177,25 +238,60 @@ namespace olympia
                 // Get the dispatchers used to dispatch the target.
                 // Find a ready-to-go dispatcher
                 auto & dispatchers = dispatchers_[target_unit];
-                for(auto & disp : dispatchers)
-                {
-                    if(disp->canAccept())
-                    {
-                        disp->acceptInst(ex_inst_ptr);
-                        ++unit_distribution_[target_unit];
-                        ++(unit_distribution_context_.context(target_unit));
-                        ++(weighted_unit_distribution_context_.context(target_unit));
-
-                        ex_inst_ptr->setStatus(Inst::Status::DISPATCHED);
-                        ILOG("Sending instruction: "
-                             << ex_inst_ptr << " to " << disp->getName());
-                        dispatched = true;
-
-                        break;
+                // so we have a map here that checks for which valid dispatchers for that instruction target pipe
+                // map needs to be:
+                // "int": [alu0, alu1, alu2, alu3]
+                // "mul": [alu0, alu1]
+                // disp->getName()
+                // ex_inst_ptr->getTarget()
+                if(ex_inst_ptr->getUnit() != InstArchInfo::TargetUnit::LSU){
+                    auto pipe = ex_inst_ptr->getPipe();
+                    auto it = pipe_map_.find(pipe);
+
+                    if (it != pipe_map_.end()) {
```

Comment:
Annnnnddd... what if it does == end()?

Reply:
Resolved in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233)